content/toolsAndResources/nge/ngeBlocks.md

```diff
@@ -734,15 +1411,119 @@ This function converts the input value from degrees to radians. This node can su
+
+<H3Image title="Dot" image="/img/tools/nge/dot.jpg" alt="Dot node"/>
+This node will calculate the cosine of the angle between two Euclidean vectors. This node supports Vector2, Vector3, and Vector4 data types.
+
+#### Debug Infos
+- **Call count** shows the number of times this node is called by nodes further into the graph. This is the total number of calls from all subsequent nodes. 
+- **Execution count** shows the number of times this node is executed. This value will affect the performance of the graph. Depending on the value of *Evaluate context* on nodes further in the graph the values for *Execution count* may be lower than the value for *Call count*. 
+
+#### Inputs
+- **left** provides the first vector for the operation.
+- **right** provides the second vector for the operation.
+
+#### Outputs
+- **output** represents the cosine of the angle between the left input and the right input as a value of Float type.
+
+<H3Image title="Length" image="/img/tools/nge/length.jpg" alt="Length node"/>
+This node will calculate the magnitude of a vector. This node supports Vector2, Vector3, and Vector4 data types.
+
+#### Debug Infos
+- **Call count** shows the number of times this node is called by nodes further into the graph. This is the total number of calls from all subsequent nodes. 
+- **Execution count** shows the number of times this node is executed. This value will affect the performance of the graph. Depending on the value of *Evaluate context* on nodes further in the graph the values for *Execution count* may be lower than the value for *Call count*. 
+
+#### Inputs
+- **value** is the vector input from which magnitude will be calculated.
+
+#### Outputs
+- **output** represents the magnitude of the input vector as a value of Float type.
+
 <H3Image title="Normalize" image="/img/tools/nge/normalizeNode.jpg" alt="Normalize node"/>
 This function transforms the input vector into a unit vector of length 1.0. This node can support Vector2, Vector3, and Vector4 types. The type for the node connected to the input of this node determines the type for the node output. 
 
+#### Debug Infos
+- **Call count** shows the number of times this node is called by nodes further into the graph. This is the total number of calls from all subsequent nodes. 
+- **Execution count** shows the number of times this node is executed. This value will affect the performance of the graph. Depending on the value of *Evaluate context* on nodes further in the graph the values for *Execution count* may be lower than the value for *Call count*. 
+
 #### Inputs
 - **input** is connected to a node providing a vector to be used in the function.
 
 #### Outputs
 - **output** is the unit vector of length 1.0 transformed from the input vector in a type that matches the input value type. 
 
+<H3Image title="Rotate2D" image="/img/tools/nge/rotate2D.jpg" alt="Rotate 2D node"/>

```

Comment:
Pictures is missing.

Reply:
link capitalization wrong. File named "rotate2d.jpg". Changing line 1513